crates/git/src/commit.rs

```diff
@@ -75,6 +80,107 @@ pub fn parse_git_diff_name_status(content: &str) -> impl Iterator<Item = (&str,
     })
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct ParsedCommitMessage {
+    pub message: SharedString,
+    pub permalink: Option<url::Url>,
+    pub pull_request: Option<crate::hosting_provider::PullRequest>,
+    pub remote: Option<GitRemote>,
+}
+
+impl ParsedCommitMessage {
+    pub fn new(
+        sha: String,
+        message: String,
+        remote_url: Option<&str>,
+        provider_registry: Option<Arc<GitHostingProviderRegistry>>,
+    ) -> Self {
+        if let Some((hosting_provider, remote)) = provider_registry
+            .and_then(|reg| remote_url.and_then(|url| parse_git_remote_url(reg, url)))
+        {
+            let pull_request = hosting_provider.extract_pull_request(&remote, &message);
+            Self {
+                message: message.into(),
+                permalink: Some(
+                    hosting_provider
+                        .build_commit_permalink(&remote, BuildCommitPermalinkParams { sha: &sha }),
+                ),
+                pull_request,
+                remote: Some(GitRemote {
+                    host: hosting_provider,
+                    owner: remote.owner.into(),
+                    repo: remote.repo.into(),
+                }),
+            }
+        } else {
+            Self::from_message(message)
+        }
+    }
+    pub fn from_message(message: String) -> Self {
+        Self {
+            message: message.into(),
+            ..Default::default()
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct CommitSummary {
+    pub sha: SharedString,
+    pub subject: SharedString,
+    /// This is a unix timestamp
+    pub commit_timestamp: i64,
+    pub author_name: SharedString,
+    pub has_parent: bool,
+}
+
+impl PartialEq for CommitSummary {
+    fn eq(&self, other: &Self) -> bool {
+        self.sha == other.sha
+    }
+}
+impl Eq for CommitSummary {}
+impl Hash for CommitSummary {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.sha.hash(state);
+    }
+}
```

Comment:
I get what these custom impls are going for, but I'd prefer to use the derived ones that account for all fields (if we need PartialEq/Eq/Hash) at all:
- it's very unlikely we are comparing `CommitSummary` in a real-world context where the savings from just comparing the SHAs matter
- it's much more likely we'd want to compare `CommitSummary` in tests where we want to actually check that all the fields match